TrackletGraph.py

```diff
@@ -167,9 +170,32 @@ def populate_bitwidths(mem,hls_dir): # FIXME this information should be parsed f
         for item in diskList:
             disk = max(disk,mem.inst.find(item))
         if mem.mtype == "VMStubsTE":
-            mem.bitwidth = 22 if mem.inst.find("L1") else 16 # FIXME
-        elif mem.mtype == "AllStubs":
+            phi_region = mem.inst.split("PHI")[1][0]
+            position = mem.inst.split("_")[1][:2]        
+            # Determine if TE outer/inner/overlap. MAKE THIS PRETTIER
+            # VMStubTEInner: 22/23 bits
+            if (position == 'L1' or position == 'L3' or position == 'D1' or position == 'D3') and (phi_region in ['A','B','C','D','E','F','G','H']):
+                mem.bitwidth = 22
+            elif position == 'L2' and phi_region in ['I','J','K','L']:
+                mem.bitwidth = 22
+            elif position == 'L5' and phi_region in ['A','B','C','D','E','F','G','H']:
+                mem.bitwidth = 23
+            #VMStubTEInner Overlap: 22 bits
+            elif (position == 'L1' or position == 'L2') and (phi_region in ['Q','R','S','T','W','X','Y','Z']):
+                mem.bitwidth = 22
+            #VMStubTEOuter: 16/17 bits
```

Comment:
Can VMStubTEOuter logic be simpler? i.e.
if position == 'L4' or position == 'L6':
  mem.bitwidth = 17
else
  mem.bitwidth = 16

Reply:
see comment above